internal/manifest/live_installer.go

```diff
@@ -0,0 +1,280 @@
+		repos:      filterRepos(repos, name),
+		kernelName: kernelName,
+		product:    product,
+		version:    version,
+	}
+	buildPipeline.addDependent(p)
+	m.addPipeline(p)
+	return p
+}
+
+// TODO: refactor - what is required to boot and what to build, and
+// do they all belong in this pipeline?
+func (p *LiveInstaller) liveInstallerBootPackageSet() []string {
+	packages := []string{
+		"grub2-tools",
+		"grub2-tools-extra",
+		"grub2-tools-minimal",
+		"efibootmgr",
+	}
+
+	switch p.platform.GetArch() {
+	case platform.ARCH_X86_64:
+		packages = append(packages,
+			"grub2-efi-x64",
+			"grub2-efi-x64-cdboot",
+			"grub2-pc",
+			"grub2-pc-modules",
+			"shim-x64",
+			"syslinux",
+			"syslinux-nonlinux",
+		)
+	case platform.ARCH_AARCH64:
+		packages = append(packages,
+			"grub2-efi-aa64-cdboot",
+			"grub2-efi-aa64",
+			"shim-aa64",
+		)
+	default:
+		panic(fmt.Sprintf("unsupported arch: %s", p.platform.GetArch()))
+	}
+
+	return packages
+}
+
+func (p *LiveInstaller) getBuildPackages() []string {
+	packages := p.liveInstallerBootPackageSet()
+	packages = append(packages,
+		"rpm",
+		"lorax-templates-generic",
+	)
+	return packages
+}
+
+func (p *LiveInstaller) getPackageSetChain() []rpmmd.PackageSet {
+	packages := p.liveInstallerBootPackageSet()
+	if p.Biosdevname {
+		packages = append(packages, "biosdevname")
+	}
+	return []rpmmd.PackageSet{
+		{
+			Include:      append(packages, p.ExtraPackages...),
+			Repositories: p.repos,
+		},
+	}
+}
+
+func (p *LiveInstaller) getPackageSpecs() []rpmmd.PackageSpec {
+	return p.packageSpecs
+}
+
+func (p *LiveInstaller) serializeStart(packages []rpmmd.PackageSpec, _ []container.Spec) {
+	if len(p.packageSpecs) > 0 {
+		panic("double call to serializeStart()")
+	}
+	p.packageSpecs = packages
+	if p.kernelName != "" {
+		p.kernelVer = rpmmd.GetVerStrFromPackageSpecListPanic(p.packageSpecs, p.kernelName)
+	}
+}
+
+func (p *LiveInstaller) serializeEnd() {
+	if len(p.packageSpecs) == 0 {
+		panic("serializeEnd() call when serialization not in progress")
+	}
+	p.kernelVer = ""
+	p.packageSpecs = nil
+}
+
+func (p *LiveInstaller) serialize() osbuild.Pipeline {
+	if len(p.packageSpecs) == 0 {
+		panic("serialization not started")
+	}
+	pipeline := p.Base.serialize()
+
+	pipeline.AddStage(osbuild.NewRPMStage(osbuild.NewRPMStageOptions(p.repos), osbuild.NewRpmStageSourceFilesInputs(p.packageSpecs)))
+	pipeline.AddStage(osbuild.NewBuildstampStage(&osbuild.BuildstampStageOptions{
+		Arch:    p.platform.GetArch().String(),
+		Product: p.product,
+		Variant: p.Variant,
+		Version: p.version,
+		Final:   true,
+	}))
+	pipeline.AddStage(osbuild.NewLocaleStage(&osbuild.LocaleStageOptions{Language: "en_US.UTF-8"}))
+
+	rootPassword := ""
+	rootUser := osbuild.UsersStageOptionsUser{
+		Password: &rootPassword,
+	}
+
+	usersStageOptions := &osbuild.UsersStageOptions{
+		Users: map[string]osbuild.UsersStageOptionsUser{
+			"root": rootUser,
+		},
+	}
+
+	pipeline.AddStage(osbuild.NewUsersStage(usersStageOptions))
```

Comment:
You should ideally use `osbuild.GenUsersStage()` instead.

Reply:
I thought the `osbuild.GenUsersStage` was (mostly) meant to go from the customization format to the stage options so I omitted it here. I'll replace it.

Reply:
I think that we don't have strong guidelines around this, but the function is meant for generating stage options from internal abstraction of the user and group definitions (which is agnostic to osbuild). I'm open to discuss what we want to do in general in such cases, I just suggested this change since we were trying to (ideally) not use stage options outside of the osbuild package. :man_shrugging:

Reply:
Then I will use `osbuild.GenUsersStage` :)

Reply:
I know :roll_eyes: :innocent:

Reply:
Even after saying it twice, the `GenUsersStage` isn't particularly friendly to explicitly empty passwords. This'll be a followup.